src/devmem.c

```diff
@@ -27,16 +27,13 @@ static int mmap_new(lua_State *L)
 	dbg("pg_size: 0x%x", pg_size);
 
 	m = (struct mmap*) lua_newuserdata(L, sizeof(struct mmap));
```

Comment:
I think this line needs to be moved after the parameter checking too, as this is the one that pushes a new value onto the stack.

Reply:
You are right, I've fixed it. However, when I tested it, it worked.